fix: LEAP-864: Fix Unit, Integration and E2E tests on LSO

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



Reason for Change:
The `htx-misc` bucket, previously used for storing files critical to our E2E and Integration tests, was deleted, causing these tests to fail. This deletion disrupted our testing pipeline.

What Does This Fix?
To address this issue, we've migrated the necessary files to an alternate bucket. This change involves updating URLs across our test suites to point to the new bucket location. The modification ensures our E2E, Integration, and Unit tests run smoothly, restoring the reliability of our testing processes.

Testing Instructions:
To verify the changes, follow these steps for each test category:

**Unit Tests:**

Run `yarn test:unit` to execute the unit tests. Ensure all tests pass without errors.

**Integration Tests:**

First, initialize the standalone editor by running `yarn lsf:serve`.
Then, execute `yarn test:integration` to run the integration tests. Confirm all tests complete successfully.

**E2E Tests:**

Start the LSO docker environment as a prerequisite for E2E tests.
Run yarn `test:e2e` to initiate the E2E testing process. Verify that all tests run and pass as expected.

### What level of testing was included in the change?
_(check all that apply)_
- [x] e2e
- [x] integration
- [x] unit

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
Merge branch 'develop' of https://github.com/HumanSignal/label-studio into develop

juliosgarbi committed:
fix: LEAP-864: Fix Unit, Integration and E2E tests on LSO

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8617775501

bmartel commented on web/libs/editor/tests/e2e/tests/audio/audio-webaudio-decoder.test.js:
```diff
@@ -83,7 +83,7 @@ Scenario('Check if regions are selected', async function({ I, LabelStudio, AtAud
   await AtAudioView.waitForAudio();
 
-  I.waitForDetached('loading-progress-bar', 10);
+  I.waitForDetached('loading-progress-bar', 30);
```
Do we need to have timeout this high? If so, why has it changed this drastically? Is it somehow related to the change in file storage location?

juliosgarbi replied:
This test failed a couple of times in my local environment due to a timeout, likely because the audio file took longer to load. Since the audio file wasn't loaded after 10 seconds, it resulted in failure. I've adjusted the timeout to ensure that this task doesn't yield a false negative result.

bmartel commented on web/libs/editor/tests/e2e/tests/regression-tests/image-zoom-position.test.js:
```diff
@@ -182,6 +182,9 @@ Data(relativeCoordsFF).Scenario('Zoomed image should keep center image in center
     await AtOutlinerPanel.dragResizerBy(shiftX, 0, AtOutlinerPanel.resizeRight, steps);
   }
+
+  I.wait(1);
```
Is this necessary, also what changed to require an additional wait to previous?

juliosgarbi replied:
This task was failing intermittently, likely because the stage had not been updated at the moment we were retrieving the stage's center position. By adding I.wait(1), the issue stopped occurring.

bmartel reviewed (approved):
Couple questions which I believe are related and if we can understand those changes I think it is fine, overall nice work 🎉

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8631945300)
> Successfully merged:  60 files changed, 649 insertions(+), 202 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-864'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8631945300

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8631976621
